app/models/tag_migration.rb

```diff
@@ -14,4 +14,25 @@ class TagMigration < ActiveRecord::Base
   def mark_as_deleted
     update!(deleted_at: DateTime.current)
   end
+
+  def aggregated_tag_mappings
+    tag_mappings_grouped_by_content_base_path.reduce([]) do |accumulator, aggregation|
+      accumulator << AggregatedTagMapping.new(content_base_path: aggregation.first, tag_mappings: aggregation.last)
+    end
+  end
+
+private
+
+  def tag_mappings_grouped_by_content_base_path
```

Comment:
This method seems to be repeated in 2 models, what do you think we could do here to avoid that?